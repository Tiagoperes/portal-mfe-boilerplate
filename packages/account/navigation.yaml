+ root ~ root.account (/account):
  + environments (/environments):
    search: string
  + organization (/organization):
  + workspaces (/workspaces):
    search: string
    type: string ('ALL' | 'OWNED')
