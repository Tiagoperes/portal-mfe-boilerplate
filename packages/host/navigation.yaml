# HOME
+ root (/):
  # ACCOUNT
  + account (/account/*):
  # STUDIOS
  + studios (/studios/*):
