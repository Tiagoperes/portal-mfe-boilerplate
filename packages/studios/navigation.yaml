+ root ~ root.studios (/studios):
  like: string
  limit: number
  + studio (/{studioId}):
    + stacks (/stacks):
      tech: array (string[])
      + stack (/{stackId}):
        + starters (/starters):
          + starter (/{starterId}):
