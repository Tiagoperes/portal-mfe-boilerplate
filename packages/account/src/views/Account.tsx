import { Breadcrumb } from 'containers/Breadcrumb'
import { ViewPropsOf } from 'navigation'

export const Account = ({ route }: ViewPropsOf<'root'>) => (
  <>
    <Breadcrumb />
    <p>Account</p>
    <a href={route.organization.$link()}>Organization</a>
    <a href={route.environments.$link({ search: 'minha busca de ambiente' })}>
      Environments
    </a>
    <a
      href={route.workspaces.$link({
        search: 'minha busca de workspace',
        type: 'ALL',
      })}
    >
      Workspaces
    </a>
  </>
)
