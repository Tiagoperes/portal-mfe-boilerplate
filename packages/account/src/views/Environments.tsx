import { Breadcrumb } from 'containers/Breadcrumb'
import { ViewPropsOf } from 'navigation'

export const Environments = ({
  route,
  params: { search },
}: ViewPropsOf<'root.environments'>) => (
  <>
    <Breadcrumb />
    <p>Environments ({search})</p>
    <a href={route.$parent.$link()}>Go back to Account</a>
  </>
)
