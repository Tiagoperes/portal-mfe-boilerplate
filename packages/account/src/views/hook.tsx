import { useNavigationContext } from 'navigation'
import { ReactElement, useState } from 'react'

export function useContent() {
  const [content, setContent] = useState<ReactElement | null>(null)

  useNavigationContext(context => {
    context
      .when('root', async props => {
        const { Account } = await import('./Account')
        setContent(<Account {...props} />)
      })
      .when('root.environments', async props => {
        const { Environments } = await import('./Environments')
        setContent(<Environments {...props} />)
      })
      .when('root.workspaces', async props => {
        const { Workspaces } = await import('./Workspaces')
        setContent(<Workspaces {...props} />)
      })
      .when('root.organization', async props => {
        const { Organization } = await import('./Organization')
        setContent(<Organization {...props} />)
      })
      .whenNotFound(() => setContent(<h1>404: Not Found</h1>))
  })

  return content
}

// Default export is required by Module Federation
// eslint-disable-next-line import/no-default-export
export default useContent
