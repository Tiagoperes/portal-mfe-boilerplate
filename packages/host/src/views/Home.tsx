import { overlay } from '@stack-spot/portal-layout'
import { Dictionary, useTranslate } from '@stack-spot/portal-translate'
import { ViewPropsOf } from 'navigation'

const dictionary = {
  en: {
    studios: 'Go to Studios',
    account: 'Go to Account',
    alert: 'This is an alert',
    confirm: 'This is a confirm',
    panel: 'This is a right panel',
    openAlert: 'Open alert',
    openConfirm: 'Open confirm',
    showPanel: 'Show Right Panel',
  },
  pt: {
    studios: 'Ir para Studios',
    account: 'Ir para Conta',
    alert: 'Isso é um alerta',
    confirm: 'Isso é um confirm',
    panel: 'Isso é um painel',
    openAlert: 'Abrir alerta',
    openConfirm: 'Abrir confirm',
    showPanel: 'Mostrar painel à direita',
  },
} satisfies Dictionary

export const Home = ({ route }: ViewPropsOf<'root'>) => {
  const t = useTranslate(dictionary)
  return (
    <>
      <p>Home</p>
      <p>
        <a href={route.studios.$link()}>{t.studios}</a>
      </p>
      <p>
        <a href={route.account.$link()}>{t.account}</a>
      </p>
      <p>
        <button
          onClick={() =>
            overlay.alert({
              title: 'Alert',
              message: t.alert,
              showButton: false,
            })
          }
        >
          {t.openAlert}
        </button>
        <button
          onClick={async () => {
            await overlay.confirm({
              title: 'Confirm',
              message: t.confirm,
            })
          }}
        >
          {t.openConfirm}
        </button>
        <button
          onClick={async () => {
            await overlay.confirm({
              title: 'Confirm',
              message: t.confirm,
              validation: 'Test',
            })
          }}
        >
          Confirm with validation
        </button>
        <button
          onClick={() =>
            overlay.showRightPanel({
              title: 'Panel',
              subtitle: t.panel,
              children: <p>Hello World!</p>,
            })
          }
        >
          {t.showPanel}
        </button>
        <button
          onClick={() =>
            overlay.showToaster({
              message: 'A successful operation.',
              type: 'success',
            })
          }
        >
          Show success toast
        </button>
        <button
          onClick={() =>
            overlay.showToaster({
              message: 'An information.',
              autoClose: false,
            })
          }
        >
          Show persistent info toast
        </button>
        <button
          onClick={() =>
            overlay.showToaster({
              message: 'This is a warning message.',
              type: 'warning',
            })
          }
        >
          Show warning toast
        </button>
        <button
          onClick={() =>
            overlay.showToaster({
              message: 'Oops! Something went wrong!',
              type: 'error',
            })
          }
        >
          Show error toast
        </button>
      </p>
    </>
  )
}
