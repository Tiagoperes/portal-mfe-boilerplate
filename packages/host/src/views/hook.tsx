import { useNavigationContext } from 'navigation'
import { ReactElement, useState } from 'react'

export function useContent() {
  const [content, setContent] = useState<ReactElement>()

  useNavigationContext(context => {
    context
      // Home
      .when('root', async props => {
        const { Home } = await import('./Home')
        setContent(<Home {...props} />)
      })
      // Account
      .whenSubrouteOf('root.account', async props => {
        const AccountView = (await import('account/View')).default
        setContent(<AccountView {...props} />)
      })
  })

  return content
}
