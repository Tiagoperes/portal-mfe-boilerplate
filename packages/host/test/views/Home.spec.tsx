import { CSSToCitricAdapter } from '@stack-spot/portal-theme'
import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
import { render } from '@testing-library/react'
import { root } from 'navigation'
import { describe, expect, it } from 'vitest'
import { Home } from 'src/views/Home'

describe('Tests of Home page', () => {
  it('Should match the snapshot', () => {
    const queryClient = new QueryClient()
    const { container } = render(
      <CSSToCitricAdapter>
        <QueryClientProvider client={queryClient}>
          <Home route={root} />
        </QueryClientProvider>
      </CSSToCitricAdapter>,
    )
    expect(container).toMatchSnapshot()
  })
})
