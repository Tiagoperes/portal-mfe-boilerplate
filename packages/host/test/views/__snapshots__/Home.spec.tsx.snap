// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tests of Home page > Should match the snapshot 1`] = `
<div>
  <p>
    Home
  </p>
  <p>
    <a
      href="/#/studios"
    >
      Go to Studios
    </a>
  </p>
  <p>
    <a
      href="/#/account"
    >
      Go to Account
    </a>
  </p>
  <p>
    <button>
      Open alert
    </button>
    <button>
      Open confirm
    </button>
    <button>
      Confirm with validation
    </button>
    <button>
      Show Right Panel
    </button>
    <button>
      Show success toast
    </button>
    <button>
      Show persistent info toast
    </button>
    <button>
      Show warning toast
    </button>
    <button>
      Show error toast
    </button>
  </p>
</div>
`;
